pytestsalt/fixtures/daemons.py

```diff
@@ -629,6 +629,122 @@ def session_salt_syndic(request,
 
 
 @pytest.yield_fixture
+def salt_proxy_before_start():
+    '''
+    This fixture should be overridden if you need to do
+    some preparation and clean up work before starting
+    the salt-proxy and after ending it.
+    '''
+    # Prep routines go here
+
+    # Start the salt-proxy
+    yield
+
+    # Clean routines go here
+
+
+@pytest.yield_fixture
+def salt_proxy_after_start(salt_proxy):
+    '''
+    This fixture should be overridden if you need to do
+    some preparation and clean up work after starting
+    the salt-proxy and before ending it.
+    '''
+    # Prep routines go here
+
+    # Resume test execution
+    yield
+
+    # Clean routines go here
+
+
+@pytest.fixture
+def salt_proxy(request,
+               salt_master,
+               proxy_id,
+               proxy_config,
+               salt_proxy_before_start,  # pylint: disable=unused-argument
+               proxy_log_prefix,
+               cli_proxy_script_name,
```

Comment:
The above `*proxy*` fixtures also need to be defined. Most of them in `pytestsalt.fixtures.config` I believe.